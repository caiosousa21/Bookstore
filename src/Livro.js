import React, { Component } from 'react';
import Card from './Card'
import Botoes from './Botoes'

class Livro extends Component{
    render(){
        return(
            <div className='Livro'>
            <Card/>
            <Botoes/>
            </div>
        )
    }
}
export default Livro;