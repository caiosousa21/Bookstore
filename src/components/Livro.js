import React, { Component } from 'react';
import Card from './Card'
import Botoes from './Botoes'

function Livro(props) {
        return(
            <div className='Livro'>
            <Card titulo={props.livro.titulo} autor={props.livro.autor} editora={props.livro.editora} data={props.livro.data}/>
           
            <Botoes livro={ ''} clickAdicionar={''}/>
            </div>
        )
    
}
export default Livro;