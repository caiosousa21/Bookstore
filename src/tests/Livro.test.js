import React from 'react';
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Livro from '../components/Livro';

Enzyme.configure({ adapter: new Adapter() })

describe('Livro', () => {
    test('Renderizando Livro', () => {
        const wrapper = shallow(<Livro />);
        expect(wrapper.exists()).toBe(true);
    }),
    it('Testando interface com snapshots',()=>{
        const livro={
            titulo:'fakeLivro', 
            autor: 'fakeAutor', 
            editora:'fakeEditora',
            data:1
        }
        const tree = renderer.create(<Livro livro={livro}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
