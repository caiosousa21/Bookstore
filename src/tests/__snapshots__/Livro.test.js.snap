// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Livro Testando interface com snapshots 1`] = `
<div
  className="Livro"
>
  <div
    className="Card"
  >
    <div
      className="Capa"
    >
      <p
        className="PCapa"
      >
        Capa
      </p>
    </div>
    <p
      className="TituloLivro"
    >
      fakeLivro
    </p>
    <p
      className="Autor"
    >
      fakeAutor
    </p>
    <p
      className="Editora"
    >
      fakeEditora
    </p>
    <p
      className="Data"
    >
      1
    </p>
  </div>
  <div
    className="Botoes"
  >
    <button
      className="BtnDescr"
    >
      Descr
    </button>
    <button
      className="BtnAdd"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;
